pip_custom_platform/util.py

```diff
@@ -3,9 +3,59 @@
 
 import contextlib
 import os
+import platform
+import re
 import shutil
 import tempfile
 
+if True:
+    # pylint: disable=import-error
+    import distutils.util
+
+
+def default_platform_name():
+    """Guess a sane default platform name.
+
+    On OS X and Windows, just uses the default platform name. On Linux, uses
+    information from the `platform` module to try to make something reasonable.
+    """
+    def sanitize(string):
+        return re.sub('[^a-z0-9_]', '_', string.lower())
+
+    def major_version(string):
+        """Grabs the major version component of a version string."""
+        return string.split('.', 1)[0]
+
+    if platform.system() == "Linux":
+        dist, version, codename = platform.linux_distribution()
+        dist = re.sub('linux$', '', dist.lower()).strip()
+        release = major_version(version)
+
+        if dist == 'red hat enterprise linux server':
+            dist = 'rhel'
+
+        # it would be nice to do this on Debian as well, but the platform
+        # module doesn't correctly detect the codename
+        if dist == 'ubuntu':
+            release = codename
+
+        # rolling releases that have no version numbers
+        # (these won't be super useful)
+        if dist == 'arch':
+            release = 'unknown'
+
+        return 'linux_{dist}_{release}_{arch}'.format(
+            dist=sanitize(dist),
+            release=sanitize(release),
+            arch=sanitize(platform.machine()),
+        )
+    else:
+        # XXX: it would be great to call pip.pep425tags.get_platform() here,
+        # but importing that can undo our distutils.util monkeypatching
+        return (
+            distutils.util.get_platform().replace('.', '_').replace('-', '_')
+        )
```

Comment:
I think distlib (already a dependency) should have an implementation of the tags you can steal